### PR TITLE
Lesson 1.3

### PR DESCRIPTION
**Pull request by stephskas**
initial commit for Lesson 1.3 working with React Props, React State, handler functions, and callback handlers

### CODE REVIEW

File: src/AddTodoForm.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  function handleAddTodo(event){
+    event.preventDefault();
+    console.log(todoTitle);
+    props.onAddTodo(todoTitle);
+    event.target.reset();
+  }
+  return (
+    <form onSubmit={(e) => handleAddTodo(e)}>
+      <label htmlFor="todoTitle">Title</label>
+      <input type="text" name="title" onInput={(e) => setTodoTitle(e.target.value)} />
```
Comment (royemosby) [inline src/AddTodoForm.js:15]: Very good of you to build a controlled form! Although this is not in the instructions for 1.3, taking this approach gives you a lot more flexibility when it comes to submission. Your use of `onInput` is correct but `onChange` is more common. Read more here in the [dev docs](https://beta.reactjs.org/reference/react-dom/components/select#props)

File: src/App.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import React, { useState } from 'react';
+import AddTodoForm from './AddTodoForm';
+import TodoList from './TodoList';
+
+
+function App() {
+  const [newTodo, setNewTodo] = useState("");
+  return (
+   <>
```
Comment (royemosby) [inline src/App.js:9]: Document fragment is a good choice

File: src/TodoListItem.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import React from "react";
+
+function TodoListItem(props) {
+  return (
+    <li key={props.key}>{props.todo.title}</li>
```
Comment (royemosby) [inline src/TodoListItem.js:5]: List item, in this context, does not need a key as this component does not create multiple elements. Its parent, TodoList is where the key needs to be and you already have that in place. Please remove this one.

**Review by royemosby — APPROVED** [review]
Overall very well done! You have incorporated some additional concepts on top of those provided in the lessons. You do not have to make the one change on this branch since you are up to lesson 1.7 by this point.

Be careful when working multiple branches in the workplace as you may be required to sequentially apply changes to multiple branches to maintain a clean history. Git can handle a rippling change across branches but the workflows get quite complex when balancing 3 or more branches. No need to make changes to this branch. Please incorporate the key removal with further PRs.

File: src/AddTodoForm.js (hunk 1 above)
Comment (stephskas) [reply src/AddTodoForm.js:15]: Roy, thank you - I really appreciate your time and feedback. Was looking forward all day to reading your review. And I will go through all your comments and follow your recommendations.

File: src/TodoListItem.js (hunk 3 above)
Comment (stephskas) [reply src/TodoListItem.js:5]: Ah ok! Got it. Thank you!

**Pull request merged**